test/solvers/advanced/forward_backward_test.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Test for the forward-backward solver."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+# External
+import pytest
+import numpy as np
+
+# Internal
+import odl
+from odl.solvers import forward_backward_pd
+from odl.util.testutils import all_almost_equal, example_element
+
+# Places for the accepted error when comparing results
+PLACES = 8
+
+
+def test_forward_backward_inpit_handeling():
+    """Test to see that input is handled correctly.
+    """
```

Comment:
At least in this case, the function name already conveys the same information as the subsequent docstring. If the test is too complex to be described in a function name, a comment can clarify stuff. One can safely assume that tests are exclusively read by developers, and only when they're failing. So the test name should state what it tests for such that it is easier to pin down the code that is causing the failure. And one can assume a certain level of sophistication of the one looking at the test code -- a minimal amount of comments of non-obvious code is sufficient.